### PR TITLE
[catalog] Introduce catalog managed

### LINKED ISSUE #3935: [Catalog] Introduce catalog crud.
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Introduce catalog crud.

### Usage Scenario

managed catalog.

### Related issues

https://github.com/apache/incubator-streampark/issues/3912

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Mrart**
## What changes were proposed in this pull request

Issue Number: close #3935 

## Brief change log

introduce catalogn managed

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.
no 
This change is already covered by existing tests, such as *(please describe tests)*.

no

This change added tests and can be verified as follows:

- *Added *CatalogServiceTest to verify the change.*
- *Manually verified the change by testing locally.* -->

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (no)

### CODE REVIEW

**Review by SbloodyS — COMMENTED** [review]
I noticed that this was a subtask of SPIP, but I didn't find the design detail or the purpose of the code in the PR and related issues. Please fill them in first. @Mrart

**Comment by Mrart** [comment]
> I noticed that this was a subtask of SPIP, but I didn't find the design detail or the purpose of the code in the PR and related issues. Please fill them in first. @Mrart
In this link
https://github.com/apache/incubator-streampark/issues/3912

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/FlinkCatalog.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+public class FlinkCatalog implements Serializable {
+
+    @TableId(type = IdType.AUTO)
+    private Long id;
+
+    private Long teamId;
+
+    private String catalogName;
+
+    private CatalogType catalogType;
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/FlinkCatalog.java:50]: How to save  CatalogType type into the database? Can you provide a test?

Comment (Mrart) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/FlinkCatalog.java:50]: I have test in ‘CatalogServiceTest’.

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**